### ISSUE TITLE
Enable ethernet networking devices

### ISSUE THREAD

**Issue by frno7** [frno7/linux#19]
The PlayStation 2 [expansion bay](https://en.wikipedia.org/wiki/PlayStation_2_Expansion_Bay) and later PlayStation 2 models with built-in hardware support the [ethernet](https://en.wikipedia.org/wiki/Ethernet) networking technology. Device driver starting points are [drivers/ps2/smap.c](https://github.com/frno7/linux/blob/ps2-v5.0/drivers/ps2/smap.c) and [drivers/ps2/smaprpc.c](https://github.com/frno7/linux/blob/ps2-v5.0/drivers/ps2/smaprpc.c), that are not quite working in the latest versions of the kernel. These two drivers should probably be combined into a single device driver.

In the meantime, USB network adapters can be used as an alternative: review the wiki article on [networking for the PlayStation 2](https://github.com/frno7/linux/wiki/Networking-for-the-PlayStation-2). See #17, #11 and #18.

**Comment by sp193** [comment]
The SMAP itself is not faultless. FYI, If you ever manage to make it go fast: https://github.com/ps2dev/ps2sdk/commit/dc918c3e511df53a5cc187015d0eaf5046966bb6#diff-a275d0f23cde1dd3f6c9d427a99f73be
It starts to exhibit stability problems once frames start transferring above 7MB/s.

**Comment by frno7** [comment]
Oh, many thanks @sp193, that was very helpful to learn.

**Comment by Arch91** [comment]
So... How it goes at this forefront?)
Do it requires a special IOP module? If yes, is this module united for both FAT and SLIM or there have to be a two (three? - for DTL...) of them?
At the first message there is a note that
> are not quite working in the latest versions of the kernel

but there was never even a chance to test it...

**Comment by frno7** [comment]
Yes, we’d need an IOP module for Ethernet. I’m hoping we could have the same network driver support all PlayStation 2 hardware variants, since that’d be easier for people, to not have to worry about which driver to use.

> > are not quite working in the latest versions of the kernel
>
> but there was never even a chance to test it...

4.x and 5.x Linux kernels have a lot of generic network subsystem changes, and so some updates are needed to compile the old drivers. Plus, the Sony BIOS isn’t available in these Linux kernels any more.

**Comment by Arch91** [comment]
Hey. An ingeniously silly idea :p Did you ever try to rely some (base) code creation to be handled by an AI?? The correct question to the AI, the useful codes for us :p For ex, I wonder how AI 'imagines' PS2 BIOS mapping in the modern kernels...

**Comment by DCFUKSURMOM** [comment]
AI is cringe

**Comment by frno7** [comment]
@Arch91, I don’t think the AI services of today are useful for my kernel hobby. If you want to talk about AI, I kindly suggest that you [open a new discussion](https://github.com/frno7/linux/discussions) about it, because this issue is about Ethernet networking support.

**Comment by frno7** [comment]
I’ve just written a detailed wiki tutorial on how to configure [networking for the PlayStation 2](https://github.com/frno7/linux/wiki/Networking-for-the-PlayStation-2).
